### DISCUSSION TITLE
Dev build of podman 4.2.0 (win) doesn't install for me

### DISCUSSION THREAD

**Discussion by n1hility** [containers/podman#15133] [General]
podman 4.2.0-dev doesn't install for me
![image](https://user-images.githubusercontent.com/73980661/181225871-28dda501-05de-40ba-846e-4313d9775e24.png)

_Originally posted by @rorygilfillan-absa in https://github.com/containers/podman/issues/14593#issuecomment-1196552321_

**Comment by n1hility** [comment]
@ericis Thanks for trying! Just to confirm the 4.1.x msi does not have this issue for you? 

Do you have any other virus scanning software that might have blocked one of the actions? There is an executable installed which is ran to update the path, and if something prevents that from running you would see that.

Could you try to capture logging and past the output to a gist/pastebin to see if any additional details are revealed:

`msiexec /i podman-v4.2.0-dev.msi /l*v installpodman.txt`

Thanks!

**Reply by ericis** [reply]
Thank @n1hility . This still fails. I've add the logs here:
https://gist.github.com/ericis/fc91e10ebc1d31415b4589b750d0c96f#file-installpodman-txt

**Reply by n1hility** [reply]
Thanks for running this again @ericis  and grabbing a log. I took a look at it, and it's looking like what would happen if a virus scanner / security tool prevented it from running, Are you running anything like that?:


```
MSI (s) (14:B8) [15:52:36:090]: System Environment : Client has accepted UAC prompt
MSI (s) (14:7C) [15:52:36:231]: Note: 1: 1721 2: AddPath 3: C:\Program Files\RedHat\Podman\winpath.exe 4: add 
```

1721 indicates it could not launch the executable. If it would have failed, we would have instead received a 1722 with an error code for the failure.

If you are running an AV tool and you check its audit log you might be able to add an exclusion.

Another thing you could try is running it directly to see if it's blocked as well. I created a zip with the extracted contents here: https://github.com/n1hility/podman-builds/blob/main/zip/podman-v4.2.0-dev-extract.zip

If you extract that and manually run winpath.exe, it should prompt with a usage help dialog,

**Reply by ericis** [reply]
Yes, I got the error @n1hility. What is `winpath.exe` attempting to access? I am using CrowdStrike Falcon Sensor and cannot disable it or it seems even view the logs.

*I was able to run the 4.1.1 installer without any issue.

![image](https://user-images.githubusercontent.com/974919/183544720-02bfe2d9-0e7b-4ff4-9c06-638591235674.png)

... I am experimenting with a WSL Debian-based podman installation to work around the issue, but really need the Windows installer to work "out of the box" in order to advocate for its use and encourage more use of containers.

**Reply by n1hility** [reply]
Ah ok, so it's likely blocking it, interesting that it doesn't prompt you. Falcon is one of the machine-learning-based scanners. In addition to matching signatures, these tools guess on the safety based on a trained model fed with known malware. Unfortunately, this is false positive prone. In particular with small executables: (a bad enough problem that golang added a faq https://go.dev/doc/faq#virus)

There was a recent enhancement to this to open a post-install tutorial/guide. I could do a test build without this patch to see if your setup gets less tripped up by it. 

I agree it should just work. Sorry for the inconvenience so far.  We hope to start signing executables soon, and once this occurs, it should lower the threat detection outcomes.

**Reply by n1hility** [reply]
@ericis can you try this one - this is with the post-install open patch removed, I'm curious if the winpath.exe in this version will not flag the scanner: https://github.com/n1hility/podman-builds/blob/main/podman-v4.2.0-devnoguide.msi

**Reply by ericis** [reply]
Thanks for the continued attempts @n1hility . That installer also fails as well. Log output can be [found here](https://gist.github.com/ericis/fc91e10ebc1d31415b4589b750d0c96f#file-install-podman-devnoguide-txt).

`msiexec /i podman-v4.2.0-devnoguide.msi /l*v install-podman-devnoguide.txt`

```
MSI (s) (90:E4) [16:07:40:236]: System Environment : Client has accepted UAC prompt
MSI (s) (90:2C) [16:07:40:258]: Note: 1: 1721 2: AddPath 3: C:\Program Files\RedHat\Podman\winpath.exe 4: add 
MSI (s) (90:2C) [16:07:40:258]: Note: 1: 2262 2: Error 3: -2147287038 
MSI (c) (98:FC) [16:07:40:266]: Font created.  Charset: Req=0, Ret=0, Font: Req=MS Shell Dlg, Ret=MS Shell Dlg

Error 1721. There is a problem with this Windows Installer package. A program required for this install to complete could not be run. Contact your support personnel or package vendor. Action: AddPath, location: C:\Program Files\RedHat\Podman\winpath.exe, command: add 
MSI (s) (90:2C) [16:07:41:357]: Note: 1: 2262 2: Error 3: -2147287038 
MSI (s) (90:2C) [16:07:41:357]: Product: Podman 4.2.0-dev -- Error 1721. There is a problem with this Windows Installer package. A program required for this install to complete could not be run. Contact your support personnel or package vendor. Action: AddPath, location: C:\Program Files\RedHat\Podman\winpath.exe, command: add 

Action ended 16:07:41: InstallFinalize. Return value 3.
MSI (s) (90:2C) [16:07:41:362]: Note: 1: 2265 2:  3: -2147287035 
```

![image](https://user-images.githubusercontent.com/974919/183750879-f69dc2f9-5f4f-4292-833b-6c3225aade20.png)

**Reply by ericis** [reply]
Let me know if you'd like me to retest anything @n1hility

**Reply by n1hility** [reply]
@ericis hey eric, sorry for the long delay. I have an early build of a new signed installer if you would like to give that a spin:
https://github.com/n1hility/podman-builds/blob/main/podman-4.2.0-unofficial-setup.exe

**Reply by ericis** [reply]
Will do. I'll reply by end of day (US EST)

**Reply by ericis** [reply]
Apologies for the delayed response @n1hility . This installer didn't quite work in the end, but installed successfully...

- [x] Installed successfully
- [x] `podman machine init` success
  - [x] Fedora download
  - [x] Fedora installation and configuration
- [x] `podman machine start`

    ```
    Starting machine "podman-machine-default"
    
    This machine is currently configured in rootless mode. If your containers
    require root permissions (e.g. ports < 1024), or if you run into compatibility
    issues with non-podman clients, you can switch using the following command:
    
            podman machine set --rootful
    
    API forwarding listening on: npipe:////./pipe/docker_engine
    
    Docker API clients default to this address. You do not need to set DOCKER_HOST.
    Machine "podman-machine-default" started successfully
    ```

- [ ] `podman images`

    ```
    podman images
    Error: error creating host key callback function for C:\Users\[USER_NAME]\.ssh\known_hosts: open C:\Users\[USER_NAME]\.ssh\known_hosts: The system cannot find the file specified.
    ```

**Reply by n1hility** [reply]
@ericis thanks for sharing and testing ! I am glad that the installation portion worked for you. Sorry about the ssh errors, this build was produced from the lastet main build tree, which was partway through an unrelated ssh refactor. The 4.3 release will complete the refactor and not have this issue. As a a workaround you can create an empty known_hosts file if you want to try it out.
